### PR TITLE
[libsoundio] Add support for linux backends

### PR DESCRIPTION
**Pull request by joe-noel-dev**
Adds features to build against optional backends (ALSA, JACK or PulseAudio).

### CODE REVIEW

**Comment by joefocusrite** [comment]
Issue #16864

File: ports/libsoundio/portfile.cmake [hunk 1]
```diff
@@ -14,6 +14,24 @@ vcpkg_from_github(
...
+    set(ENABLE_JACK ON)
+else()
+    set(ENABLE_JACK OFF)
+endif()
+
+if("pulseaudio" IN_LIST FEATURES)
+    set(ENABLE_PULSEAUDIO ON)
+else()
+    set(ENABLE_PULSEAUDIO OFF)
+endif()
```
Comment (PhoebeHui) [inline ports/libsoundio/portfile.cmake:33]: Could you use [vcpkg_check_features](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/vcpkg_check_features.md) instead?

File: ports/libsoundio/vcpkg.json [hunk 2]
```diff
@@ -0,0 +1,22 @@
+{
+  "name": "libsoundio",
+  "version-string": "2.0.0-3",
+  "port-version": 1,
```
Comment (PhoebeHui) [inline ports/libsoundio/vcpkg.json:4]:
```suggestion
  "version": "2.0.0",
  "port-version": 4,
```

**Review by PhoebeHui — CHANGES_REQUESTED** [review]
Thanks for the PR!

**Comment by JackBoosY** [comment]
Waiting for #16904 merged.

**Review by JackBoosY — APPROVED** [review]

**Comment by JackBoosY** [comment]
Need test all features.

**Comment by JackBoosY** [comment]
Test passed on `x86-windows` / `x64-windows-static` / `x64-linux`.

File: versions/l-/libsoundio.json [hunk 3]
```diff
@@ -1,5 +1,15 @@
...
+    {
+      "git-tree": "96de946826ca6aafdc9409f9d75f02a6d082cffb",
+      "version": "2.0.0",
+      "port-version": 4
+    },
+    {
+      "git-tree": "b8772b81312859cfef91f4161ebcf236acea836e",
+      "version-string": "2.0.0-3",
+      "port-version": 1
+    },
```
Comment (vicroms) [inline versions/l-/libsoundio.json:12]:
```suggestion
```
Please remove this entry to avoid duplicates with old version.

**Review by vicroms — CHANGES_REQUESTED** [review]

**Comment by ras0219-msft** [comment]
Thanks for the improvements!

**Pull request merged**